### PR TITLE
routerrpc: adapt to changed interfaces of routing subsystem

### PR DESCRIPTION
**Pull request by joostjager**

### CODE REVIEW

File: lnrpc/routerrpc/config_active.go [hunk 1]
```diff
@@ -40,4 +40,8 @@ type Config struct {
 	//
 	// TODO(roasbeef): assumes router handles saving payment state
 	Router *routing.ChannelRouter
+
+	// RouterBackend contains shared logic between this sub server and the
+	// main rpc server.
+	RouterBackend *RouterBackend
```
Comment (Roasbeef) [inline lnrpc/routerrpc/config_active.go]: We only need this for the self node pubkey right? I think we can pass in just that instead.

Comment (joostjager) [reply lnrpc/routerrpc/config_active.go]: In this PR it is just that one, but my follow up https://github.com/lightningnetwork/lnd/pull/1662 requires access to other parts of it.

**Comment by Roasbeef** [comment]
Needs a rebase!

**Comment by joostjager** [comment]
@Roasbeef rebased ptal

**Comment by Roasbeef** [comment]
Fails travis: 
```
    lnd_test.go:107: lnd finished with error (stderr):
        exit status 2
        panic: reflect: call of reflect.Value.Set on zero Value
```

**Comment by joostjager** [comment]
@Roasbeef bug fixes, ptal

**Pull request merged**